packages/wdio-config/src/lib/ConfigParser.ts

```diff
@@ -142,6 +142,13 @@ export default class ConfigParser {
         if (exclude.length > 0) {
             this._config.exclude = this.setFilePathToFilterOptions(exclude, this._config.exclude!)
         }
+
+        /**
+         * Parse cucumberOpts.names into an array, if defined
+         */
+        if (this._config.cucumberOpts?.names != null && typeof this._config.cucumberOpts.names == 'string') {
+            this._config.cucumberOpts.names = this._config.cucumberOpts?.names.split(',')
+        }
```

Comment:
I wonder if we can move this here: https://github.com/webdriverio/webdriverio/blob/306ce22d67656e5e6e9a8941ea2ff368b05b6dc2/packages/wdio-cucumber-framework/src/index.ts#L75 to have less cucumber specific code in the configparser

Reply:
It can be in a different place but the main issue I fix is that `cucumberOpts.names` is assigned a `string` when it should be assigned a `string[]`.  I don't see how your proposal addresses that.

Reply:
I am fine with the way you are solving the problem, just saying that the location of the code is not ideal. We should just move it to the `@wdio/cucumber-framework` package.

Reply:
Oh, OK. I misunderstood you. Yeah, it could make sense to move it there (I tested it locally and it worked fine). However, this solution would be just a quick fix for this specific property. Isn't the root of the problem that the config parser is, in general, not parsing array options properly? In the lines above my change, there are a few cases dealing with array vs string. Also, in this file, there is already Cucumber specific code.